test/e2e/e2e.go

```diff
@@ -130,10 +139,9 @@ var _ = Describe("validation", func() {
 		var metallbCRExisted bool
 
 		BeforeEach(func() {
-			metallb = &metallbv1beta1.MetalLB{}
-			err := loadMetalLBFromFile(metallb, consts.MetalLBCRFile)
+			var err error
+			metallb, err = getMetalLB()
```

Comment:
i think this can be `metallb, err := getMetalLB()`

Reply:
No. If we do that we will get a local instance of metallb.
We want to use the metallb declared in the Context, so we can use the same one in the AfterEach.

Reply:
i was referring to the error variable

Reply:
Doing:
metallb, err := getMetalLB()
will also create a new local metallb variable.

We can either do:
   metallb, err := getMetalLB()   - here we have 2 new variables override all vars from previous scope (including the metallb defined in Context)
or
  metallb, err = getMetalLB() - this is using 2 vars defined previously
Unfortunately there is no way to use a previously declared metallb, and create a new err :(

Reply:
you are right. thanks!